spec/classes/lightblue/application/datamgmt_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'spec_helper'
+
+describe 'lightblue::application::datamgmt' do
+  let(:hiera_config){ 'spec/fixtures/hiera/hiera.yaml' }
+
+  let :facts do
+    {
+      :architecture => 'x86_64',
+      :osfamily => 'RedHat',
+      :operatingsystemrelease => '7.0'
+    }
+  end
+
+  context 'with default params' do
+    it do
+      should contain_package('lightblue-data-mgmt') \
+        .with_ensure('latest') \
+        .with_require(['Class[Lightblue::Yumrepo::Lightblue]', 'Class[Lightblue::Eap]'])
+    end
+
+    it do
```

Comment:
I think it's reasonable if it's a guideline and not a requirement.  I honestly don't see much benefit unless the set of conditions tested are large and independent.  And even then, it's more useful only for the initial development, unless the class under testing is britile and prone to breaking due to other class changes.

For example, I have helped write an XSLT unit testing framework that collects all failures before reporting a test failed.  This was useful because XSLT is a best and it was easy to break many things with one change somewhere else that didn't seem related.